app/views/questionnaire.py

```diff
@@ -121,6 +121,10 @@ def post_block(eq_id, form_type, collection_id, group_id, group_instance, block_
     else:
         _update_questionnaire_store(current_location, form)
         next_location = path_finder.get_next_location(current_location=current_location)
+
+        if next_location is None:
```

Comment:
I think if None is returned that's a valid reason of saying there is no next location therefore you're at the end. That said we might want to do something in the case where the current_location isn't a valid location